model/cicd/registry.yaml

```diff
@@ -75,3 +75,78 @@ groups:
         brief: >
           The type of the task within a pipeline.
         examples: ["build", "test", "deploy"]
+      - id: cicd.pipeline.result
+        type:
+          members:
+            - id: success
+              value: success
+              brief: "The pipeline run finished successfully."
+              stability: experimental
+            - id: failure
+              value: failure
+              brief: >-
+                The pipeline run did not finish successfully, eg. due to a compile error or a failing test.
+                Such failures are usually detected by non-zero exit codes of the tools executed in the pipeline run.
+              stability: experimental
+            - id: error
+              value: error
+              brief: >-
+                The pipeline run failed due to an error in the CICD system, eg. due to the worker being killed.
+              stability: experimental
+            - id: timeout
+              value: timeout
+              brief: "A timeout caused the pipeline run to be interrupted."
+              stability: experimental
+            - id: cancelled
+              value: cancelled
+              brief: "The pipeline run was cancelled, eg. by a user manually cancelling the pipeline run."
+              stability: experimental
+            - id: skipped
+              value: skipped
+              brief: "The pipeline run was skipped, eg. due to a precondition not being met."
+              stability: experimental
+        stability: experimental
+        brief: >
+          The result of a pipeline run.
+        examples: ["success", "failure", "timeout", "skipped"]
+      - id: cicd.worker.class
```

Comment:
is there an example of when this attribute is necessary? 

It seems to introduce some classification where members are not mutually exclusive (container vs pod), but also don't describe worker type enough (e.g. would we need to differentiate container runtime).

I wonder if we can capture useful information in some other form.

E.g. if we have worker-level metrics, we'd associate them with a set of resource attributes covering all you need to know about workers.

Reply:
I thought that it would be good to be able to subdivide the `cicd.worker.count`.

I agree that vm/pod/container might not be the most useful subdivision here and that this kind of information is better left to the worker-level metrics.

I will remove it for now to not block this PR.
When we have a more concrete use cases for worker.count subdivision we can revisit this discussion.

Reply:
Done in b3326385c794f476f1c02036b1486cf6049a4c47 and cf67fd2d746574a599e7c1c86fbf7c6b7fecc696